felix/bpf/ut/nat_test.go

```diff
@@ -1896,6 +1896,260 @@ func TestNATNodePortIngressDSR(t *testing.T) {
 	Expect(v.Flags()).To(Equal(conntrack3.FlagNATFwdDsr | conntrack3.FlagNATNPFwd))
 }
 
+func TestNATNodePortDSROptout(t *testing.T) {
```

Comment:
Wonder if you could refactor a bit to make the between-packet set-up common and maybe add more comments to really spell out what you're doing.  I think that'd aid readability, especially for folks who aren't familiar with these tests
```
// Pretend that we're on node 1, set up NAT maps with a node port.
maps := mustSetUpMaps(WithNodePort(...)) 

// Simulate a packet arriving at node 1.  Expect the BFP program to encap the packet and redirect to node 2.
...

// Now pretend that we're on node 2...  Configure the NAT maps again.
...
// Feed it the packet from node 1, expect it to spot that it's a no-DSR source
// and record that in conntrack before forwarding to a local pod.

// Now pretend we've got the response packet form the pod at egress from the pod...

```